lib/inventory/controller.go

```diff
@@ -491,6 +503,13 @@ func (c *Controller) handlePingRequest(handle *upstreamHandle, req pingRequest)
 }
 
 func (c *Controller) handleHeartbeatMsg(handle *upstreamHandle, hb proto.InventoryHeartbeat) error {
+	// If a goodbye was received that indicated active resource
+	// clean up should occur, then ignore processing any heartbeats
+	// that may have been emitted after the goodbye.
+	if handle.goodbye != nil && handle.goodbye.DeleteResources {
+		return nil
+	}
```

Comment:
nit: I could go either way on this, but my gut feeling is that this condition is added brittleness/complexity without much added upside.  I'm partial to the idea of the goodbye not changing anything about the state-machine's behavior until the stream actually exits.  That feels like a simpler/more reliable system to me.  I do see the upside of avoiding unnecessary writes tho, so take it or leave it.